microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -64,66 +65,89 @@ def __init__(
             self._remote_tar_path = self.get_tool_path(
                 use_global=True
             ) / os.path.basename(self._tar_file_path)
-        else:
-            self._remote_tar_path = (
-                self.get_tool_path(use_global=True)
-                / f"{self._KERNEL_REPO_NAME}{self._KSELFTEST_TAR_PATH}"
-            )
 
         # command to run kselftests
-        if not self._tar_file_path:
-            self._command = self.get_tool_path(use_global=True) / "run_kselftest.sh"
-        else:
-            self._command = (
-                self.get_tool_path(use_global=True)
-                / "kselftest-packages/run_kselftest.sh"
-            )
+        self._kself_installed_dir = (
+            self.get_tool_path(use_global=True) / "kselftest-packages"
+        )
+
+        self._command = self._kself_installed_dir / "run_kselftest.sh"
 
     # install common dependencies
     def _install(self) -> bool:
-        if not isinstance(self.node.os, CBLMariner):
+        if not (
+            isinstance(self.node.os, Ubuntu) or isinstance(self.node.os, CBLMariner)
+        ):
             raise UnsupportedDistroException(
-                self.node.os, "kselftests are supported on Mariner VMs only."
+                self.node.os, "kselftests are supported on Ubuntu and Mariner only"
             )
 
         if self._tar_file_path:
             self.node.shell.copy(PurePath(self._tar_file_path), self._remote_tar_path)
-        else:
-            # clone kernel, build kernel, then build kselftests
-            self.node.os.install_packages(
-                [
-                    "bison",
-                    "flex",
-                    "build-essential",
-                    "openssl-devel",
-                    "bc",
-                    "dwarves",
-                    "rsync",
-                    "libcap-devel",
-                    "libcap-ng-devel",
-                    "fuse",
-                    "fuse-devel",
-                    "popt-devel",
-                    "numactl-devel",
-                    "libmnl-devel",
-                    "libinput",
-                    "mesa-libgbm-devel",
-                    "glibc-static",
-                    "clang",
-                ]
+            self.node.tools[Tar].extract(
+                str(self._remote_tar_path), str(self._kself_installed_dir), sudo=True
             )
+            self._log.debug(f"Extracted tar from path {self._remote_tar_path}!")
+        else:
+            mkdir = self.node.tools[Mkdir]
+            mkdir.create_directory(str(self._kself_installed_dir))
+            if isinstance(self.node.os, Ubuntu):
+                # cache is used to speed up recompilation
+                self.node.os.install_packages(
```

Comment:
Is there a way, where we can maintain this list somewhere in json or other place in LISA, based upon the Distro and programatically install, instead of keeping track everywhere we create . import new testcases/suits to lisa.

Reply:
It's ok to add it here. You can move it to top of the file like configuration. Tests have different dependencies, and it won't be changed frequently. So it's better to track it in test code.

Reply:
Updated